contracts/interchain-token-service/src/state.rs

```diff
@@ -40,27 +43,35 @@ pub struct ChainConfig {
     pub truncation: TruncationConfig,
     pub its_address: Address,
     pub frozen: bool,
+    pub translation_contract: Addr,
 }
 
-#[cw_serde]
-pub struct TruncationConfig {
-    pub max_uint_bits: NumBits, // The maximum number of bits used to represent unsigned integer values that is supported by the chain's token standard
-    pub max_decimals_when_truncating: u8, // The maximum number of decimals that is preserved when deploying from a chain with a larger max unsigned integer
-}
-
-impl From<msg::ChainConfig> for ChainConfig {
-    fn from(value: msg::ChainConfig) -> Self {
-        Self {
+impl ChainConfig {
+    pub fn from_input(input: msg::ChainConfig, api: &dyn cosmwasm_std::Api) -> Result<Self, Error> {
```

Comment:
```suggestion
    pub fn from_msg(input: msg::ChainConfig, api: &dyn cosmwasm_std::Api) -> Result<Self, Error> {
```
or `from_config`, or even just calling it `new`

Reply:
why not change the `From` to `TryFrom` instead?

Reply:
it can't be TryFrom because we need the additional api argument